CLI - introduce clean command

### Description

Introduce clean command for CLI.

ATM CLI has commands `build`, `dev`, `test`, ... But there is no `clean` command.

As a workaround I run `mvn clean` to get fresh start, e.g. when running dev mode and changing Quarkus version.

I think clean could be invoked for maven and gradle, not sure about JBang.
For JBang based app there is no `quarkus dev` support for it, so maybe similar error message approach could be used.
```
quarkus dev
[ERROR] ❗  Unable to launch project in dev mode: Not there yet. ;)
```


### Implementation ideas

_No response_

--- a/docs/src/main/asciidoc/cli-tooling.adoc
+++ b/docs/src/main/asciidoc/cli-tooling.adoc
@@ -503,6 +503,8 @@ quarkus build
 
 NOTE: Output will vary depending on the build tool your project is using (Maven, Gradle, or JBang).
 
+NOTE: Run `quarkus build --clean` to perform clean as part of the build.
+
 === Development mode
 
 [source,shell]
@@ -537,6 +539,7 @@ Tests paused, press [r] to resume
 
 NOTE: Output will vary depending on the build tool your project is using (Maven, Gradle, or JBang).
 
+NOTE: Run `quarkus dev --clean` to perform clean as part of the build.
 
 [[quarkus-version-compatibility]]
 [WARNING]

